myriad-scheduler/src/main/java/org/apache/myriad/configuration/NodeManagerConfiguration.java

```diff
@@ -91,6 +102,18 @@ public Double getCpus() {
     return Optional.fromNullable(cpus).or(DEFAULT_NM_CPUS);
   }
 
+  public synchronized Map<String, Long> getPorts() {
+    if (ports == null) {
+      //Good idea to have deterministic order
```

Comment:
Not explicitly needed but helpful for unit tests.

We could possibly use an ArrayList<Pair<String,Long>> but I felt it would be awkward to define in the yaml file (I wanted the ability to add additional ports for other YARN service, ie spark shuffle service).